app/services/workflow_executions/completion_service.rb

```diff
@@ -145,11 +147,45 @@ def put_output_attachments_onto_samples
         next if swe.outputs.empty?
 
         files = swe.outputs.map { |output| output.file.signed_id }
-        params = { files: }
+        params = { files:, include_activity: false }
         Attachments::CreateService.new(
           current_user, swe.sample, params
         ).execute
       end
     end
+
+    def create_activities # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+      @workflow_execution.samples_workflow_executions&.each do |swe|
+        next if swe.metadata.nil? && swe.outputs.empty?
+
+        if !swe.metadata.nil? && !swe.outputs.empty?
+          @workflow_execution.namespace.create_activity key: 'workflow_execution.automated_workflow_completion.outputs_and_metadata_written', # rubocop:disable Layout/LineLength
+                                                        parameters: {
+                                                          workflow_id: @workflow_execution.id,
+                                                          sample_id: swe.sample.id,
+                                                          sample_name: swe.sample.puid
```

Comment:
**nitpick (blocking):** Please renamed this to `sample_puid`

Reply:
Updated in 032cb1c08ee993dee5d18b74ef198b03bafe07b3